### DISCUSSION TITLE
blog/call-llm-api-in-python/

### DISCUSSION THREAD

**Discussion by giscus** [jeremy-feng/giscus#9] [Announcements]
# blog/call-llm-api-in-python/

https://fengchao.pro/blog/call-llm-api-in-python/

**Comment by Neptune-0724** [comment]
你好，我想咨询一下，在导入包“from chatllm.llmchain.llms import ChatGLM, ErnieBot, SparkBot”，存在“NameError: Field name "validate_environment" shadows a BaseModel attribute; use a different field name with "alias='validate_environment'".”的问题

**Reply by jeremy-feng** [reply]
应该是版本的问题，这是我用的版本：
```
chatllm==2023.9.26.18.14.16
dashscope==1.12.0
langchain==0.0.325
openai==0.27.8
python-dotenv==1.0.0
```

**Reply by Neptune-0724** [reply]
好的，非常感谢，将pip包修改为您提供的版本号，已经没有报错了。另外想请问这是源码吗？我对这这个教材复现，并没有得到各大模型的回复。

**Reply by jeremy-feng** [reply]
这是调用各个大语言模型的 API 的代码。如果你正确配置了 API Key 等参数，应该是可以得到回复的呀。你可以检查一下你的 API 是否正确，以及里面是否有余额（国内的这几个大模型一般是有免费额度的）
